engines/agos/event.cpp

```diff
@@ -503,6 +503,46 @@ void AGOSEngine::delay(uint amount) {
 						GUI::AboutDialog aboutDialog;
 						aboutDialog.runModal();
 					}
+				} else if (event.kbd.keycode == Common::KEYCODE_f && getGameId() == GID_WAXWORKS) {
+					if (getLanguage() == Common::EN_ANY) {
+						// Check if the necessary HitAreas exist.
+						HitArea *fightButton = findBox(117);
+						HitArea *walkButton = findBox(119);
+
+						if (fightButton && walkButton) {
+							// Hotkey is supported (HitAreas exist)
+							HitArea *ha = nullptr;
+							if (_mouseCursor == 3) {
+								// If already in fighting mode, switch to a default non-fighting cursor (0)
+								_mouseCursor = 0;
+								_needHitAreaRecalc++;
```

Comment:
This is increased in both if branches, so it can be done once outside the if statements